paimon-web-api/src/main/java/org/apache/paimon/web/api/catalog/PaimonServiceFactory.java

```diff
@@ -26,15 +26,30 @@
 
 import org.apache.commons.lang3.StringUtils;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /** Paimon service factory. */
 public class PaimonServiceFactory {
 
-    public static PaimonService createFileSystemCatalogService(String name, String warehouse) {
+    public static PaimonService createFileSystemCatalogService(
+            String name, String warehouse, Map<String, String> catalogOptions) {
+        if (catalogOptions == null) {
+            catalogOptions = new HashMap<>();
+        }
         Options options = new Options();
         options.set(CatalogProperties.WAREHOUSE, warehouse + "/" + name);
-
+        String fileSystemType = catalogOptions.get("fileSystemType");
+        if ("s3".equals(fileSystemType)) {
```

Comment:
What's the behavior of other filesystem type?

Reply:
No other options need to be set for the local file system, and HDFS is not currently supported